aptos-move/framework/supra-framework/sources/multisig_voting.spec.move

```diff
@@ -0,0 +1,427 @@
+        // ensures result == spec_get_proposal_state(voting_forum_address, proposal_id, voting_forum);
+    }
+
+    spec get_proposal_creation_secs {
+        include AbortsIfNotContainProposalID<ProposalType>;
+        let voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        let proposal = table::spec_get(voting_forum.proposals, proposal_id);
+        ensures result == proposal.creation_time_secs;
+    }
+
+    spec get_proposal_expiration_secs {
+        include AbortsIfNotContainProposalID<ProposalType>;
+        ensures result == spec_get_proposal_expiration_secs<ProposalType>(voting_forum_address, proposal_id);
+    }
+
+    spec get_execution_hash {
+        include AbortsIfNotContainProposalID<ProposalType>;
+        let voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        let proposal = table::spec_get(voting_forum.proposals, proposal_id);
+        ensures result == proposal.execution_hash;
+    }
+
+    spec get_min_vote_threshold {
+        include AbortsIfNotContainProposalID<ProposalType>;
+        let voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        let proposal = table::spec_get(voting_forum.proposals, proposal_id);
+        ensures result == proposal.min_vote_threshold;
+    }
+
+    spec get_votes {
+        include AbortsIfNotContainProposalID<ProposalType>;
+        let voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        let proposal = table::spec_get(voting_forum.proposals, proposal_id);
+        ensures result_1 == proposal.yes_votes;
+        ensures result_2 == proposal.no_votes;
+    }
+
+    spec is_resolved {
+        include AbortsIfNotContainProposalID<ProposalType>;
+        let voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        let proposal = table::spec_get(voting_forum.proposals, proposal_id);
+        ensures result == proposal.is_resolved;
+    }
+
+    spec schema AbortsIfNotContainProposalID<ProposalType> {
+        proposal_id: u64;
+        voting_forum_address: address;
+        let voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        aborts_if !table::spec_contains(voting_forum.proposals, proposal_id);
+        aborts_if !exists<VotingForum<ProposalType>>(voting_forum_address);
+    }
+
+    spec schema IsProposalResolvableAbortsIf<ProposalType> {
+        voting_forum_address: address;
+        proposal_id: u64;
+
+        include AbortsIfNotContainProposalID<ProposalType>;
+
+        let voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        let proposal = table::spec_get(voting_forum.proposals, proposal_id);
+        let voting_closed = spec_is_voting_closed<ProposalType>(voting_forum_address, proposal_id);
+        // Avoid Overflow
+        aborts_if voting_closed && (proposal.yes_votes <= proposal.no_votes || proposal.yes_votes + proposal.no_votes < proposal.min_vote_threshold);
```

Comment:
It doesn't seem it reflects our logic wrt `yes` and `no` votes. We would abort if `voting_closed && !(proposal.yes_votes >= proposal.min_vote_threshold)` isn't it?

Reply:
You are right, I have updated it!